HW4/HW4.py

```diff
@@ -0,0 +1,191 @@
+
+    binned_mean_sq_diffs = (binned_means - res_mean) ** 2
+
+    title = f"Mean Square Diff of {res_col_name} VS Binned {pred_col_name}: "
+    secondary_y_title = f"Square Diff of {res_col_name}"
+    if weighted:
+        weights = binned_counts / len(pred_values)
+        binned_mean_sq_diffs = binned_mean_sq_diffs * weights
+        title = "Weighted " + title
+        secondary_y_title = "Weighted" + secondary_y_title
+
+    res_mean_sq_diff = np.nansum(binned_mean_sq_diffs) / 10
+
+    title += str(res_mean_sq_diff)
+
+    bin_width = bin_edges[1] - bin_edges[0]
+    bin_centers = bin_edges[1:] - bin_width / 2
+
+    fig = make_subplots(specs=[[{"secondary_y": True}]])
+    fig.add_trace(go.Bar(x=bin_centers, y=binned_counts), secondary_y=False)
+    fig.add_trace(go.Scatter(x=bin_centers, y=binned_mean_sq_diffs), secondary_y=True)
+
+    fig.update_layout(title_text=title)
+    fig.update_xaxes(title_text=f"{pred_col_name}")
+    fig.update_yaxes(title_text=f"{pred_col_name} Distribution", secondary_y=False)
+    fig.update_yaxes(title_text=secondary_y_title, secondary_y=True)
+    fig.show()
+
+    return
+
+
+# HW4.py
+# Ask for user input
+# Input data file name: suspectheartrates.csv
+# Input comma separated predictor column names:
+# Input response column name:
+
+data_df = pandas.read_csv("suspectheartrates.csv")
+# pred_col_names = ["heartbpm", "age", "sex"]
+# res_col_name = "veracity"
+pred_col_names = ["veracity", "age", "sex"]
+res_col_name = "heartbpm"
+
+res_is_cat = is_column_cat(data_df, res_col_name)
+for pred_col_name in pred_col_names:
+    if res_is_cat:
+        if is_column_cat(data_df, pred_col_name):
+            # Pred is cat and Res is cat, plot heatmap
+            plot_heatmap(data_df, pred_col_name, res_col_name)
+        else:
+            # Pred is cont and Res is cat, plot grouped hist
+            # Calculate p-value and t-score, Logistic Regression
+            plot_grouped_hist(
+                data_df,
+                cont_col_name=pred_col_name,
+                cat_col_name=res_col_name,
+                n_bins=50,
+            )
+            pass
+    else:
+        if is_column_cat(data_df, pred_col_name):
+            # Pred is cat and Res is cont, plot grouped hist
+            plot_grouped_hist(
+                data_df,
+                cont_col_name=res_col_name,
+                cat_col_name=pred_col_name,
+                n_bins=50,
+            )
+            pass
+        else:
+            # Pred is cont and Res is cont, plot scatter with trendline
+            # Calculate p-value and t-score, Linear Regression
+            plot_scatter(data_df, pred_col_name, res_col_name)
+            plot_mean_sq_diff(
+                data_df, pred_col_name, res_col_name, weighted=False, n_bins=15
+            )
+            plot_mean_sq_diff(
+                data_df, pred_col_name, res_col_name, weighted=True, n_bins=30
+            )
+            pass
```

Comment:
I don't see you doing the Random Forest part, calculating p t values, and generating a table with variables and rankings

Reply:
I just commit a new version but I have not generated a table with variables and ranking.